nav2_costmap_2d/include/nav2_costmap_2d/footprint_subscriber.hpp

```diff
@@ -30,32 +30,47 @@ class FootprintSubscriber
 public:
   FootprintSubscriber(
     nav2_util::LifecycleNode::SharedPtr node,
-    const std::string & topic_name);
+    const std::string & topic_name,
+    const double & footprint_timeout);
 
   FootprintSubscriber(
     rclcpp::Node::SharedPtr node,
-    const std::string & topic_name);
+    const std::string & topic_name,
+    const double & footprint_timeout);
 
   FootprintSubscriber(
     const rclcpp::node_interfaces::NodeBaseInterface::SharedPtr node_base,
     const rclcpp::node_interfaces::NodeTopicsInterface::SharedPtr node_topics,
     const rclcpp::node_interfaces::NodeLoggingInterface::SharedPtr node_logging,
-    const std::string & topic_name);
+    const rclcpp::node_interfaces::NodeClockInterface::SharedPtr node_clock,
+    const std::string & topic_name,
+    const double & footprint_timeout);
 
   ~FootprintSubscriber() {}
 
+  // Returns an oriented robot footprint.
+  // The footprint received by the subscriber has already been transformed
+  // such that all points are in the global frame.
+  // This frame is assumed to be known out of band, but typically it is "map".
```

Comment:
"hand"?

Reply:
Ah, that was from the other PR, I glanced over it. I’ll fix that up and shorten this description

Reply:
fixed